### PR TITLE
Tests and Error messages for QR IBAN Sepa Export

### PR DESCRIPTION
**Pull request by LarsSander**
https://github.com/metasfresh/metasfresh/issues/13384

### CODE REVIEW

File: backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java [hunk 1]
```diff
@@ -695,7 +694,12 @@ else if (!Check.isEmpty(line.getSwiftCode(), true))
 
 					if(isInvalidQRReference(QRReference))
 					{
-						throw new SepaMarshallerException("SEPA_ExportLine " + createInfo(line) + " has no valid QR Reference");
+						final IMsgBL msgBL = Services.get(IMsgBL.class);
+
+						final ITranslatableString message = msgBL
+								.getTranslatableMsgText(ERR_SEPA_Export_InvalidReference,createInfo(line));
+
+						throw new AdempiereException(message);
```
Comment (teosarca) [inline backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java:702]: I think u can directly call
```java
throw new AdempiereException(ERR_SEPA_Export_InvalidReference, createInfo(line));
```

File: backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java [hunk 2]
```diff
@@ -246,15 +246,10 @@ public void marshal(
...
-			final Document xmlDocument = createDocument(sepaDocument);
-			marshal(xmlDocument, out);
-		}
-		catch (final Exception e)
-		{
-			throw new AdempiereException("Error while marshaling " + sepaDocument, e);
-		}
+
+		final Document xmlDocument = createDocument(sepaDocument);
+		marshal(xmlDocument, out);
```
Comment (teosarca) [inline backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java:251]: why we are no longer catching the exception here? what's the reason behind?
might be ok, but i want to understand.

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java (hunk 2 above)
Comment (LarsSander) [reply backend/de.metas.payment.sepa/base/src/main/java/de/metas/payment/sepa/sepamarshaller/impl/SEPAVendorCreditTransferMarshaler_Pain_001_001_03_CH_02.java:251]: image

The error messages where not displayed correctly.

**Review by teosarca — APPROVED** [review]

**Pull request merged**